### ISSUE TITLE
ICE Anon Const w/ Inference Var but no error

### ISSUE THREAD

**Issue by JulianKnodt** [rust-lang/rust#137160]
### Code

Not public sorry
It occurs when I add an `unwrap_or(GenericType<3>)`, and the genericType has `where : [(); Sum::<N>()]`, and
```
pub const fn Sum<const N: usize>() -> usize {
  N * (N + 1) / 2
}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.87.0-nightly (5bc623145 2025-02-16)
```

### Error output

```
note: no errors encountered even though delayed bugs were created

note: those delayed bugs will now be shown as internal compiler errors

error: internal compiler error: Encountered anon const with inference variable args but no error reported
  |
  = note: delayed at compiler/rustc_trait_selection/src/traits/mod.rs:598:27 - disabled backtrace

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
The query stack is empty
```

</p>
</details>

**Comment by theemathas** [comment]
This code compiles fine:

```rust
#![feature(generic_const_exprs)]

struct GenericType<const N: usize> where [(); Sum::<N>()]:;

pub const fn Sum<const N: usize>() -> usize {
    N * (N + 1) / 2
}

fn foo() {
    None.unwrap_or(GenericType::<3>);
}
```

Could you provide your code that gave you the ICE?

**Comment by JulianKnodt** [comment]
I can't, the code isn't public.

**Comment by JulianKnodt** [comment]
https://github.com/JulianKnodt/rustc_ice_137160

I think having it be in two separate crates might be necessary, but not totally sure.

**Comment by jieyouxu** [comment]
This looks familiar to me... maybe duplicate of #133199?

**Comment by lqd** [comment]
@jieyouxu That's absolutely true, good spot.

Adding

```rust
#![feature(generic_const_exprs)]
#![allow(incomplete_features)]
```

to `src/bin/repro.rs` fixes the issue.

Let's close this as duplicate then.
